### PR TITLE
Refresh documentation on network connectivity

### LINKED ISSUE #91: At least to me the networking documentation is hard to follow
>Can we follow a little bit more the structure like we have for https://kubevirt.io/user-guide/#/workloads/virtual-machines/disks-and-volumes? The structure there is lent from k8s, which I personally find pretty easy to follow and user-focused.
>
>So to have 
> * Examples per configs
> * For instance also show where and how to set the model
> * For a comprehensive and up-to-date list of supported values, link to the api-reference and make sure that the api-reference comment contains supported values.
>
> It may just be me, but I find the structure of this document very confusing and it is almost impossible for me to parse it and extract the necessary information. It also looks a little bit like this documentation tries to be a mixture of api-reference and user-guide. We should leverage the fact that we have an api-reference.

from https://github.com/kubevirt/user-guide/pull/89#issuecomment-399029970

### PR DESCRIPTION
**Pull request by booxter**
The page was not in line with the rest of documentation, had multiple bugs,
incorrect information, typos, grammar issues.

This patch reworks the structure and contents of the page from the ground up,
with the emphasis on providing timely examples and capturing just the features
that are already implemented.

Fixes #91

### CODE REVIEW

**Comment by booxter** [comment]
@vladikr @rmohr please have a look, this PR rebuilds our networking docs.

**Comment by booxter** [comment]
In the latest upload, I merely resolved git conflicts.

File: workloads/virtual-machines/interfaces-and-networks.md [hunk 1]
```diff
@@ -1,90 +1,150 @@
...
+Network backends are configured in `spec.networks`. Each network must have a
+unique name. Additional fields declare which logical or physical device the
+network relates to.
 
-- a list of interfaces and
-- a list of networks each of the interfaces is connected to
+Each network definition can be referred to in no more than a single interface
+per virtual machine. If you need to connect the same virtual machine twice to
+the same physical or logical network, declare two networks with the same
+properties but unique names.
```
Comment (vladikr) [inline workloads/virtual-machines/interfaces-and-networks.md]: Perhaps, we shouldn't talk about connecting the same network to multiple interfaces now. After all, it won't work now with the pod network delegate ip approach, not sure what's the benefit of it with slirp.
I think we should get back to this point later when we will have additional networks

File: workloads/virtual-machines/interfaces-and-networks.md [hunk 2]
```diff
@@ -1,90 +1,150 @@
...
+    devices:
+      interfaces:
+        - name: red
+          slirp: {} # connect using SLIRP mode
+  networks:
+  - name: red
+    pod: {}
+```
+
+At this time, `bridge` mode doesn't support additional configuration
```
Comment (vladikr) [inline workloads/virtual-machines/interfaces-and-networks.md]: s/bridge/slirp ?

File: workloads/virtual-machines/interfaces-and-networks.md (hunk 1 above)
Comment (booxter) [reply workloads/virtual-machines/interfaces-and-networks.md]: I thought "**Note:** Currently, the only supported network type is `pod`." gives enough of a signal it's not yet possible but intended to be. I will clean it up though.

**Comment by booxter** [comment]
@vladikr please make another review round; I hope I accommodated for your comments.

**Comment by booxter** [comment]
@vladikr AFAIU this part should be resolved in scope of: https://github.com/kubevirt/user-guide/pull/84

**Comment by vladikr** [comment]
> @vladikr AFAIU this part should be resolved in scope of: #84

I see. OK, sounds good.

**Comment by vladikr** [comment]
Thanks!
